roboflow/core/workspace.py

```diff
@@ -61,6 +62,64 @@ def project(self, project_name):
 
         return Project(self.__api_key, dataset_info, self.model_format)
 
+    def active_learning(self, raw_data, inference_endpoint, upload_destination, conditionals):
+        '''
+        @params:
+            raw_data: dir = folder of images, or videos, to be processed
+            inference_endpoint: List[str, int] = name of the project
+            upload_destination: str = name of the upload project
+            conditionals: dict = dictionary of upload conditions
+        '''
+
+        inference_model = self.project(inference_endpoint[0]).version(inference_endpoint[1]).model
+        upload_project = self.project(upload_destination)
+
+        print("inference reference point: ", inference_model)
+        print("upload destination: ", upload_project)
+
+        # TODO: work with raw_data
+        # TODO: extention and globbing properties added to config
+        video_extention = ".png"
+        globbed_files = glob.glob(raw_data + '/*' + video_extention)
+
+        for index, image in enumerate(globbed_files):
+            print("*** Processing image [" + str(index + 1) + "/" + str(len(globbed_files)) + "] - " + image + " ***")
+
+            # perform inference on image
+            # TODO: mention 403 error
+            predictions = inference_model.predict(image).json()['predictions']
+            
+            # compare object and class count of predictions if enabled, continue if not enough occurances
+            if(not count_comparisons(predictions, conditionals["required_objects_count"], conditionals["required_class_variance_count"], conditionals["target_classes"])):
+                print(' [X] image failed count cases')
+                continue 
+
+            # iterate through all predictions
+            for prediction in predictions:
+
+                # check if box size of detection fits requirements
+                if not check_box_size(prediction, conditionals["minimum_size_requirement"], conditionals["maximum_size_requirement"]):
+                    print(' [X] prediction failed box size cases')
+                    continue
+
+                # compare confidence of detected object to confidence thresholds
+                # confidence comes in as a .XXX instead of XXX%
+                if(prediction['confidence'] * 100 >= conditionals["confidence_interval"][0] and 
+                    prediction['confidence'] * 100 <= conditionals["confidence_interval"][1]):
+                
+                    # filter out non-target_class uploads if enabled
+                    if(len(conditionals["target_classes"]) > 0 and
+                        prediction['class'] not in conditionals["target_classes"]):
+                        print(' [X] prediction failed target_classes')
+                        continue
+
+                    # upload on success!
+                    print(' >> image uploaded!')
+                    upload_project.upload(image, num_retry_uploads=3)
```

Comment:
Does this attempt to upload multiple instances of the image if multiple prediction boxes pass the logic? Seems like we should just track a flag in the for loop and upload once

Reply:
Nope - you had the break at the end, rad